File: dotnet/cafdemalihapa/Directories/Hosting/Feature/Tools/ToolsDirectory.cs
namespace cafdemalihapa.Directories.Hosting.Feature.Tools
{
    public class ToolsDirectory()
    {
        public string GetPath()
        {
            var directoryName = "tools";
            var featureDirectoryPath = FeatureDirectory.GetPath();
            var notesAndMessages = Path.Combine(featureDirectoryPath, directoryName);
            return notesAndMessages;
        }

        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/Tools/ToolsDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.Tools

open System.IO
open cafdemwimu.console.Directories.Hosting.Feature

type ToolsDirectory() =
    member _.GetPath() =
        let directoryName = "tools"
        let featureDirectoryPath = FeatureDirectory.GetPath()
        let notesAndMessages = Path.Combine(featureDirectoryPath, directoryName)
        notesAndMessages

    member this.Create() =
        let path = this.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore
